File: backend/app/models/tool.py
from typing import List
from pydantic import BaseModel


class ToolDefinition(BaseModel):
    id: str
    label: str
    description: str = ""
    category: str = ""
    supported_inputs: List[str]
    output_type: str
    multi_file: bool = False
    configurable: bool = False

File: frontend/types/tool.ts
import { NormalizedFileType } from "./file";

export type ToolCategory =
  | "convert"
  | "organize"
  | "optimize"
  | "extract";

export interface ToolDefinition {
  id: string;
  label: string;
  description: string;
  category: ToolCategory;
  supportedInputs: NormalizedFileType[];
  outputType: string;
  multiFile: boolean;
  configurable: boolean;
  /** Optional schema for the options panel, used later. */
  optionsSchema?: Record<string, unknown>;
}